pkg/reconciler/workload/resource/resource_controller.go

```diff
@@ -98,19 +98,21 @@ func NewController(
 
 		getSyncTargetFromKey: func(syncTargetKey string) (*workloadv1alpha1.SyncTarget, bool, error) {
 			syncTargets, err := indexers.ByIndex[*workloadv1alpha1.SyncTarget](syncTargetInformer.Informer().GetIndexer(), bySyncTargetKey, syncTargetKey)
-			if err != nil && !errors.IsNotFound(err) {
-				return nil, false, err
+			if errors.IsNotFound(err) || len(syncTargets) == 0 {
+				// check cache before deciding what to return
+				syncTargets, err = indexers.ByIndex[*workloadv1alpha1.SyncTarget](globalSyncTargetInformer.Informer().GetIndexer(), bySyncTargetKey, syncTargetKey)
+				if errors.IsNotFound(err) || len(syncTargets) == 0 {
+					// still no SyncTarget, so return empty results.
+					return nil, false, nil
+				}
 			}
-			if errors.IsNotFound(err) {
-				return nil, false, nil
+			if err != nil {
+				return nil, false, err
```

Comment:
Will we come to this line (`err != nil`) ? if `syncTargets` was `nil` above we would not come here, even though `err != nil` afaict.